plugins/optimization-detective/storage/data.php

```diff
@@ -143,21 +143,53 @@ function od_get_url_metrics_slug( array $query_vars ): string {
 /**
  * Gets the current ETag for URL Metrics.
  *
- * The ETag is a hash based on the IDs of the registered tag visitors
- * in the current environment. It is used for marking the URL Metrics as stale
- * when its value changes.
+ * Generates a hash based on the IDs of registered tag visitors, queried posts,
+ * and theme information in the current environment. This ETag is used to assess
+ * if the URL Metrics are stale when its value changes.
  *
  * @since n.e.x.t
  * @access private
  *
+ * @global WP_Query $wp_the_query            Global WP_Query instance.
+ * @global string   $_wp_current_template_id Current template ID.
+ * @global string   $template                Template file path.
+ *
  * @param OD_Tag_Visitor_Registry $tag_visitor_registry Tag visitor registry.
  * @return non-empty-string Current ETag.
  */
 function od_get_current_url_metrics_etag( OD_Tag_Visitor_Registry $tag_visitor_registry ): string {
+	if ( isset( $GLOBALS['wp_the_query']->posts ) && is_array( $GLOBALS['wp_the_query']->posts ) ) {
+		$queried_posts = array_map(
+			static function ( WP_Post $post ): array {
+				return array(
+					'ID'            => $post->ID,
+					'last_modified' => $post->post_modified_gmt,
+				);
+			},
+			$GLOBALS['wp_the_query']->posts
+		);
+	} else {
+		$queried_posts = array();
+	}
+
 	$data = array(
-		'tag_visitors' => array_keys( iterator_to_array( $tag_visitor_registry ) ),
+		'tag_visitors'  => array_keys( iterator_to_array( $tag_visitor_registry ) ),
+		'queried_posts' => $queried_posts,
+		'active_theme'  => array(
+			'template'           => get_template(),
+			'template_version'   => wp_get_theme( get_template() )->get( 'Version' ),
+			'stylesheet'         => get_stylesheet(),
+			'stylesheet_version' => wp_get_theme()->get( 'Version' ),
```

Comment:
Including the theme's version here seemed relevant since a theme update could potentially change the template content, making the URL metrics stale.

Reply:
The rationale for `template_version` is that if the current template comes from the parent theme, any updates to the parent theme should invalidate the ETag. Including the template's version seemed like a good way to ensure this dependency is accounted for.